### PR TITLE
feat: nervos crosschain | bounties/hackathons

### PR DESCRIPTION
**Pull request by chibie**
#### Frontend
- [x] Add chain to bounty creation + hack form form (staff flag)
- [x] Funder should be able to create bounty with nervos
- [x] User should be able to start / stop / submit work with nervos
- [x] Funder should be able to payout using web3

#### Backend
- [x] Add in chain sync logic
- [x] Fix fulfil flow for bounty
- [x] Change web3 type to nervos_ext
- [x] Fix Amount bug
- [x] testing

##### Refers/Fixes

#8438 

##### Testing

create bounty + submit work -> https://www.loom.com/share/933c147955854db98e54a6e3a958e69c
bounty payout -> https://www.loom.com/share/2303204b24bb4f9988485ea20a97e0c9

### CODE REVIEW

File: app/assets/v2/js/pages/bounty_detail/nervos_extension.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const payWithNervosExtension = async (fulfillment_id, to_address, vm, modal) => {
+
+    callback(null, vm.bounty.bounty_owner_address, vm.fulfillment_context.payout_tx_id);
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_detail/nervos_extension.js:3]: ^ reindent file :P

File: app/assets/v2/js/pages/bounty_detail/nervos_extension.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  
+    function callback(error, from_address, txn) {
+      if (error) {
+        _alert({ message: gettext('Unable to payout bounty due to: ' + error) }, 'danger');
+        console.log(error);
+      } else {
+        console.log(txn);
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_detail/nervos_extension.js:10]: remove log?

File: app/assets/v2/js/pages/bounty_detail/nervos_extension.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  
+        modal.closeModal();
+        const apiUrlBounty = `/api/v1/bounty/payout/${fulfillment_id}`;
+  
+        fetchData(apiUrlBounty, 'POST', payload).then(response => {
+          if (200 <= response.status && response.status <= 204) {
+            console.log('success', response);
+  
+            vm.fetchBounty();
+            _alert('Payment Successful');
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_detail/nervos_extension.js:29]: wrap it around  _alert('...', 'sucess'); ? or does this work the same way?

File: app/assets/v2/js/pages/bounty_details2.js [hunk 4]
```diff
@@ -392,12 +400,17 @@ Vue.mixin({
...
+      let vm = this;
 
       if (web3_type == 'manual') {
         tenant = 'OTHERS';
         return tenant;
       }
 
+      if (vm.bounty.web3_type == 'nervos_ext') {
+        vm.canChangeFunderAddress = true;
+      }
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details2.js:412]: ^ could we make this into an array ['nervos_ext'] cause we have one more chain coming in which would want this feature.

File: app/assets/v2/js/pages/bounty_details2.js [hunk 5]
```diff
@@ -754,7 +775,9 @@ Vue.mixin({
       switch (fulfillment.payout_type) {
         case 'qr':
         case 'manual':
+        case 'nervos_ext':
           vm.fulfillment_context.active_step = 'check_wallet_owner';
+          vm.getTenant(fulfillment.payout_type);
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details2.js:780]: why do we invoke this here ?

File: app/assets/v2/js/pages/bounty_details2.js [hunk 6]
```diff
@@ -803,6 +826,39 @@ Vue.mixin({
           vm.fulfillment_context.active_step = 'payout_amount';
           break;
       }
+    },
+    showWorkSubmitted: function(handle) {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details2.js:830]: this is not being used anywhere right ?

File: app/assets/v2/js/pages/bounty_details2.js [hunk 7]
```diff
@@ -803,6 +826,39 @@ Vue.mixin({
...
+      return false;
+    },
+    validateFunderAddress: function() {
+      let vm = this;
+
+      vm.errors = {};
+
+      if (!vm.validateNervos()) {
+        vm.$set(vm.errors, 'funderAddress', 'Please enter a valid Nervos address');
+      }
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details2.js:861]: ^ this is related to ckb -> so let's move it to nervos_extension.js

File: app/assets/v2/js/pages/bounty_details2.js [hunk 8]
```diff
@@ -803,6 +826,39 @@ Vue.mixin({
...
+      if (
+        vm.contxt.is_staff ||
+        vm.isOwner ||
+        (handle && handle == vm.contxt.github_handle)
+      ) {
+        return true;
+      }
+      return false;
+    },
+    validateNervos: function() {
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details2.js:842]: ^ this is related to ckb -> so let's move it to nervos_extension.js

File: app/assets/v2/js/pages/bounty_details2.js (hunk 5 above)
Comment (chibie) [reply app/assets/v2/js/pages/bounty_details2.js:780]: cause of L410

File: app/assets/v2/js/pages/bounty_details2.js (hunk 6 above)
Comment (chibie) [reply app/assets/v2/js/pages/bounty_details2.js:830]: oops... now i'm tryna figure out why this is here... looks like important stuff though

File: app/assets/v2/js/pages/bounty_details2.js (hunk 7 above)
Comment (chibie) [reply app/assets/v2/js/pages/bounty_details2.js:861]: i can move `validateNervos` but `validateFunderAddress` is intended to be generic

Comment (chibie) [reply app/assets/v2/js/pages/bounty_details2.js:854]: update: `nervos_extension.js` no longer exists (https://github.com/gitcoinco/web/pull/8550/commits/8174c4d64a751ddce132d893375a1659933418a5)... realised it wasn't used anywhere... i retested

File: app/assets/v2/js/pages/bounty_details2.js (hunk 6 above)
Comment (thelostone-mc) [reply app/assets/v2/js/pages/bounty_details2.js:823]: that was reverted ! i'm assuming a merge / rebase at your end when wrong 😓 
https://github.com/gitcoinco/web/pull/8790
^ could you check that PR and remove that piece of code

File: app/assets/v2/js/pages/bounty_details2.js (hunk 7 above)
Comment (thelostone-mc) [reply app/assets/v2/js/pages/bounty_details2.js:856]: wait how do you fire a backend call without `nervos_extension.js`  ? :P I can't seem to find that trigger at my end

File: app/assets/v2/js/pages/bounty_details2.js (hunk 6 above)
Comment (chibie) [reply app/assets/v2/js/pages/bounty_details2.js]: done https://github.com/gitcoinco/web/pull/8550/commits/9af812618e10373350424e16e3663fb61cc2ee05

File: app/assets/v2/js/pages/bounty_details2.js (hunk 7 above)
Comment (chibie) [reply app/assets/v2/js/pages/bounty_details2.js:844]: i used the crosschain flow in `details2.html` from L630

it fires backend call from `fulfillmentComplete()` in `bounty_details2.js` from L476

**Review by thelostone-mc — APPROVED** [review]

**Review by zacheryschiller — APPROVED** [review]

**Review by gdixon — APPROVED** [review]
All looks good to me @Chibie! Good job! 🚀

**Pull request merged**